### DISCUSSION TITLE
Take a value for "block: spacing" from "par: leading"

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4956] [Q&A] [answered]
The paragraphs in my document are separated one from another using first-line indent, and so the spacing between paragraphs should be the same as leading.

To make it easy to adjust this value, I can use a variable:

```
#let leading_and_spacing = 1em
#set par(first-line-indent: 1em, justify: true, leading: leading_and_spacing)
#show par: set block(spacing: leading_and_spacing)
```

But maybe there is an easier way, something like this? (The current version doesn't work. There is an error.)

```
#set par(first-line-indent: 1em, justify: true, leading: 1em)
#show par: set block(spacing: par.leading)
```

**Comment by johnmapeson** [comment]
Probably related, though I didn't figure out how to adapt it for my case: https://github.com/typst/typst/issues/763

**Comment by PgBiel** [comment, marked as answer]
`par.leading` (as well as any other styling queries) only works inside a `context { }` block, which means you'd have to wrap any sections of your document inside one if you want to use that set rule (or the entire document to apply to all of it). In my opinion, the variable approach is cleaner.

**Comment by tshu-w** [comment]
Something like the following works:
```typst
  show grid: it => context {
    set block(spacing: par.leading)
    it
  }
```
